proposals/ytt/003-openapi-schemas/README.md

```diff
@@ -0,0 +1,148 @@
+
+All of these flags will behave similar to the `--data-values-inspect` flag, which excludes any templates in the output.
+
+Here is a list of the similar current and proposed flags
+```bash
+# current flags
+ytt --files-inspect                 # prints involved files
+ytt --data-values-inspect           # prints values files
+ytt --data-values-inspect -ojson    # prints values as json
+
+# New flags
+ytt --data-values-schema-inspect -o openapi     # prints data values schema files as an OpenAPI Document
+
+# Additional New flags
+ytt --data-values-inspect -o openapi    # prints values files as OpenAPI Document
+ytt --data-values-schema-inspect    # prints ytt schema (as ytt schema with annotations)
+ytt --data-values-schema-inspect -o openapi-schema    # prints data values schema files as an OpenAPI Schema
+```
+
+## Examples:
+### Scenario 1: Data Values Schema file only
+`ytt -f schema.yml --data-values-schema-inspect -o openapi` will generate an OpenAPI Schema document using the Data Values Schema `schema.yml` file.
+
+### Scenario 2: Data Values Schema and Data Values (AKA Data Values Overlays)
+`ytt -f schema.yml -f values.yml --data-values-schema-inspect -o openapi` will generate an OpenAPI Schema document using **only** the Data Values Schema `schema.yml`, and will ignore the Data Values file `values.yml`.
+
+**Note**: Restricting this flag to only use Data Values Schema files results in the values from Data Values Overlays not appearing in the OpenAPI Document, despite that they may ultimately be used as Data Values during templating. However, once Data Values Schema files support default values for Arrays, Configuration Authors will be able to supply all their values as Data Values Schema, and the need for using Data Values Overlays can be replaced with using Data Value Schema overlays, thus avoiding this issue.
+
+### Scenario 3: Data Values file only
+`ytt -f values.yml --data-values-schema-inspect -o openapi` will result in a warning suggesting that a Data Values Schema file be provided.
+
+`ytt -f values.yml --data-values-inspect -o openapi` will generate an OpenAPI Schema document using both the Data Values Schema `schema.yml`, and the Data Values file `values.yml`
+
+## Use Cases
+### Inserting an OpenAPI Schema into a kapp-controller Package CR
+A kapp-controller [Package](/kapp-controller/docs/latest/packaging/#package-1) has a `valuesSchema` key where the OpenAPI Schema is placed to provide documentation on what data values may be used with ytt templated configuration. Currently, this is a manual process to create the OpenAPI Schema. This is the recommended workflow:
+1. Create an OpenAPI Document from a Data Values Schema file: `ytt -f schema.yml --data-values-schema-inspect -o openapi`.
+```yaml
+#! openapi.yml
+openapi: 3.1.0
```

Comment:
I did some [research](https://gist.github.com/cari-lynn/bc35270734e41ea17947e2f43a631494) on the differences between 3.0.x and 3.1.0. They are very differencet specs, with breaking changes. 3.1.0 was released this year, and has full compatability with JsonSchema and some nice advantages. However 3.1.0 does not have very much adoption based on [this site](https://openapi.tools/) and a look through the GitHub OpenAPI 3 topic. Because it hasn't been adopted by many tools yet if users want to use a tool to validate their OpenAPI Document they may not be able to find one yet. For this reason I think we should use 3.0.x.